pinot-common/src/main/java/org/apache/pinot/common/utils/TarGzCompressionUtils.java

```diff
@@ -198,4 +238,30 @@ public static void untarOneFile(File inputFile, String fileName, File outputFile
       throw new IOException(String.format("Failed to find file: %s in: %s", fileName, inputFile));
     }
   }
+
+  public static long streamCopyWithRateLimiter(InputStream inputStream, FileOutputStream outputStream, long rateLimit)
```

Comment:
As suggested offline, the function name / config use is confusing. We need to choose better naming to clearly expose the 2 flavors

- non-streaming download (existing)
- streaming download
  -- with rate limiter
  -- without rate limiter

Reply:
Changed namings